embrace-android-sdk/src/main/java/io/embrace/android/embracesdk/telemetry/EmbraceTelemetryService.kt

```diff
@@ -0,0 +1,77 @@
+package io.embrace.android.embracesdk.telemetry
+
+import io.embrace.android.embracesdk.capture.metadata.EmbraceMetadataService
+import io.embrace.android.embracesdk.internal.spans.toEmbraceAttributeName
+import java.util.Locale
+
+/*
+    Service for tracking usage of public APIs, and different internal metrics about the app.
+ */
+internal class EmbraceTelemetryService(
+    private val okHttpReflectionFacade: OkHttpReflectionFacade = OkHttpReflectionFacade()
+) {
+
+    private val usageCountMap = mutableMapOf<String, Int>()
+    private val appAttributesMap = mutableMapOf<String, String>()
+
+    /*
+        Tracks the usage of a public API by name. Adds a suffix for easier identification. Replaces whitespaces with underscores.
+    */
+    fun onPublicApiCalled(name: String) {
+        val suffixedName = "emb.usage.${
+        name.toLowerCase(Locale.ENGLISH)
+            .trim()
+            .replace(" ", "_")
+        }"
+        usageCountMap[suffixedName] = (usageCountMap[suffixedName] ?: 0) + 1
+    }
+
+    /*
+        Returns a map with every telemetry value. For now, it's just usage counts of public APIs.
+     */
+    fun getTelemetryAttributes(): Map<String, String> {
+        val telemetryMap = mutableMapOf<String, String>()
+
+        telemetryMap.putAll(getUsageCountTelemetry())
+
+        telemetryMap.putAll(getAppAttributes())
+
+        return telemetryMap
```

Comment:
`return getUsageCountTelemetry().plus(getAppAttributes())`

Would avoid returning an instance of a mutable map

Reply:
nice catch 🙌

Reply:
I already added this: 
```
      "attributes": {
        "emb.sequence_id": "1",
        "emb.private": "true",
        "emb.type": "SESSION",
        "emb.okhttp3": "true",
        "emb.okhttp3_on_classpath": "4.9.3",
        "emb.kotlin_on_classpath": "1.4.32",
        "emb.is_emulator": "__EMBRACE_TEST_IGNORE__",
        "emb.usage.log_breadcrumb": "1",
        "emb.usage.set_user_email": "1",
        "emb.usage.set_user_identifier": "1",
        "emb.usage.set_user_persona": "1",
        "emb.usage.set_username": "1"
      },
```

To the SessionMessageTest session-end.json golden file. Do you think we should add some other test? Do we need to call every public API, or would this be enough?

Reply:
That seems reasonable for coverage